### PR TITLE
Fix private remote address with RFC 1918

### PR DESCRIPTION
**Pull request by johnstyle**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Fix private remote address with RFC 1918 ( https://tools.ietf.org/html/rfc1918 )
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -
| How to test?  | -

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9158)

### CODE REVIEW

**Review by mickaelandrieu — APPROVED** [review]

**Comment by mickaelandrieu** [comment]
Hi @johnstyle, how can we test it?

**Comment by PierreRambaud** [comment]
We maybe can use `filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE)`? http://php.net/manual/en/filter.filters.flags.php

**Comment by MathiasReker** [comment]
What is status on this?

**Comment by PierreRambaud** [comment]
@Mreker Still waiting for author

**Comment by johnstyle** [comment]
You can not test it easily it's a rule for load balancing server...

**Comment by Quetzacoalt91** [comment]
(Base branch updated for `develop`.)
If we can't test easily, we should at least add some unit tests for the IP check.

File: classes/Tools.php [hunk 1]
```diff
@@ -355,9 +355,8 @@ public static function getRemoteAddr()
             $_SERVER['HTTP_X_FORWARDED_FOR'] = $headers['X-Forwarded-For'];
         }
 
-        if (isset($_SERVER['HTTP_X_FORWARDED_FOR']) && $_SERVER['HTTP_X_FORWARDED_FOR'] && (!isset($_SERVER['REMOTE_ADDR'])
-            || preg_match('/^127\..*/i', trim($_SERVER['REMOTE_ADDR'])) || preg_match('/^172\.16.*/i', trim($_SERVER['REMOTE_ADDR']))
-            || preg_match('/^192\.168\.*/i', trim($_SERVER['REMOTE_ADDR'])) || preg_match('/^10\..*/i', trim($_SERVER['REMOTE_ADDR'])))) {
+        if (isset($_SERVER['HTTP_X_FORWARDED_FOR']) && $_SERVER['HTTP_X_FORWARDED_FOR']
+            && (!isset($_SERVER['REMOTE_ADDR']) || preg_match('/^(10|127|172\.(1[6-9]|2\d|3[0-1])|192\.168)\./', trim($_SERVER['REMOTE_ADDR'])))) {
```
Comment (PierreRambaud) [inline classes/Tools.php:359]: Can you use `filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_NO_PRIV_RANGE | FILTER_FLAG_NO_RES_RANGE)` instead?

**Comment by PierreRambaud** [comment]
Fixed in #17785
Thanks @johnstyle

**Pull request closed without merging**